### PR TITLE
Generate documentation of SQL tables

### LINKED ISSUE #980: Generate documentation for introspection tables
It would be nice if we can automate the generation of the documentation for the introspection tables. One cheap solution could be to generate that out of the define_schema macro.

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fix #980

### CODE REVIEW

File: crates/storage-query-datafusion/src/deployment/schema.rs [hunk 1]
```diff
@@ -15,8 +15,8 @@ use crate::table_macro::*;
...
 
 define_table!(deployment(
-    id: DataType::LargeUtf8,
-    ty: DataType::LargeUtf8,
-    endpoint: DataType::LargeUtf8,
-    created_at: DataType::Date64,
+    id "The ID of the service deployment.": DataType::LargeUtf8,
+    ty "The type of the endpoint. Either `http` or `lambda`.": DataType::LargeUtf8,
+    endpoint "The address of the endpoint. Either HTTP URL or Lambda ARN.": DataType::LargeUtf8,
+    created_at "Timestamp indicating the deployment registration time.": DataType::Date64,
```
Comment (AhmedSoliman) [inline crates/storage-query-datafusion/src/deployment/schema.rs:21]: The syntax adds a lot of inline noise. Maybe an alternative is to move this to be a comment on every field instead.

Comment (tillrohrmann) [reply crates/storage-query-datafusion/src/deployment/schema.rs:21]: I like this idea. It is a bit similar to how clap does it for the command line options, right?

Comment (tillrohrmann) [reply crates/storage-query-datafusion/src/deployment/schema.rs:21]: I will try to finish this PR while slinkydeveloper is on vacation to avoid PR rot.

**Comment by tillrohrmann** [comment]
I've pushed an update to incorporate @AhmedSoliman's idea to use Rust docs instead.

**Comment by AhmedSoliman** [comment]
Looks a lot better. The only thing I'd suggest (potentially for a future PR) is to consider an alternative to the match per table name when generating the docs.

**Review by AhmedSoliman — APPROVED** [review]

**Comment by tillrohrmann** [comment]
> Looks a lot better. The only thing I'd suggest (potentially for a future PR) is to consider an alternative to the match per table name when generating the docs.

Good point. I'll look whether there is an easy solution for it.

**Comment by tillrohrmann** [comment]
I briefly looked into using [linkme](https://crates.io/crates/linkme) or [inventory](https://crates.io/crates/inventory) for automatically register the different table schemas at compile/before `main` time. Unfortunately, both seem to not properly work on MacOS because of https://github.com/dtolnay/linkme/issues/61. A way to solve it would be to require at least `lto = "thin"` for the `dev` profile so that `xtask` will see the different elements. I stopped at this point and will leave it as a future task to properly figure it out.

**Pull request merged**